wandb/sdk/lib/apikey.py

```diff
@@ -179,20 +217,32 @@ def write_netrc(host: str, entity: str, key: str) -> Optional[bool]:
                 key_suffix, len(key_suffix)
             )
         )
-        return None
-    try:
-        normalized_host = urlparse(host).netloc.split(":")[0]
-        netrc_path = get_netrc_file_path()
-        wandb.termlog(
-            f"Appending key for {normalized_host} to your netrc file: {netrc_path}"
+        return False
+
+    normalized_host = urlparse(host).netloc
+    netrc_path = get_netrc_file_path()
+    netrc_access = check_netrc_access(netrc_path)
+
+    if not netrc_access.write_access or not netrc_access.read_access:
+        wandb.termwarn(
+            f"Cannot access {netrc_path}. In order to persist your API key,"
+            + "\nGrant read & write permissions for your user to the file,"
+            + '\nor specify a different file with the environment variable "NETRC={new_netrc_path}".'
```

Comment:
The explicit `\n` isn't necessary; we assume elsewhere that the user's terminal wraps lines.

Some grammar fixes:
* In "In order to persist your API key, Grant ...", the Grant shouldn't be capitalized
* There shouldn't be a comma after "grant ... permissions ... to the file" ("In order to __, grant or specify." not "In order to __, grant, or specify")

Some formatting fixes:
* Explicit "+" isn't necessary in Python; strings are concatenated by juxtaposition `"abc" "xyz" == "abcxyz"`
* The braces in `NETRC={new_netrc_path}` could be mistaken for string interpolation; could you write something like `NETRC=<new path>` instead?
* Could you break the string so that each line fits into 80 characters?